conjure-java-core/src/test/resources/primitive-collections.yml

```diff
@@ -0,0 +1,9 @@
+types:
+  definitions:
+    default-package: com.palantir.product
+    objects:
+      PrimitiveExample:
```

Comment:
I made this separate of `ListExample` so that I could control the generator settings separately which would allow me to test that my serialization optimization continued to respect various Jackson settings without mucking with existing tests.